nomos-mix/message/src/packet.rs

```diff
@@ -0,0 +1,250 @@
+use crate::Error;
+use serde::{Deserialize, Serialize};
+use sphinx_packet::constants::NODE_ADDRESS_LENGTH;
+
+/// A packet that contains a header and a payload.
+/// The header and payload are encrypted for the selected recipients.
+/// This packet can be serialized and sent over the network.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Packet {
+    header: Header,
+    // This crate doesn't limit the payload size.
+    // Users must choose the appropriate constant size and implement padding.
+    payload: Vec<u8>,
+}
+
+/// The packet header
+#[derive(Debug, Serialize, Deserialize)]
+struct Header {
+    /// The ephemeral public key for a recipient to derive the shared secret
+    /// which can be used to decrypt the header and payload.
+    ephemeral_public_key: x25519_dalek::PublicKey,
+    // TODO: Length-preserved layered encryption on RoutingInfo
+    routing_info: RoutingInfo,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct RoutingInfo {
+    // TODO: Change this to `is_final_layer: bool`
+    // by implementing the length-preserved layered encryption.
+    // It's not good to expose the info that how many layers remain to the intermediate recipients.
+    remaining_layers: u8,
+    // TODO:: Add the following fields
+    // header_integrity_hamc
+    // additional data (e.g. incentivization)
```

Comment:
Are we gonna asses this in the next PRs? If not, lets open issues.

Reply:
Additional data is not determined yet in the spec. Marcin is mainly working on designing this mechanism. I'll open an issue for this but it will take some time to kick off the dev.

Reply:
header_integrity_hmac will be added at the next PR. That's essential for encryption/decryption.